### PR TITLE
RavenDB-14201: Add Test for restoring legacy backups via MigrateDatabaseOffline

### PR DESCRIPTION
**Pull request by garayx**

### CODE REVIEW

**Review by grisha-kotler — APPROVED** [review]

File: src/Raven.Server/Web/ServerRequestHandler.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+{
+    public abstract class ServerRequestHandler : RequestHandler
+    {
+        public override void Init(RequestHandlerContext context)
+        {
+            base.Init(context);
+
+            var topologyEtag = GetLongFromHeaders(Constants.Headers.TopologyEtag);
+            if (topologyEtag.HasValue && Server.ServerStore.HasTopologyChanged(topologyEtag.Value))
+                context.HttpContext.Response.Headers[Constants.Headers.RefreshTopology] = "true";
```
Comment (ayende) [inline src/Raven.Server/Web/ServerRequestHandler.cs:13]: why is this required?

Comment (grisha-kotler) [reply src/Raven.Server/Web/ServerRequestHandler.cs:13]: It's for the `ClusterRequestExecutor` in the client. So we can update its topology if needed.

File: src/Raven.Server/Web/System/AdminDatabasesHandler.cs [hunk 2]
```diff
@@ -705,7 +705,6 @@ public async Task RestoreDatabase()
                     throw new ArgumentException($"No enough free space to restore a backup. Required space {desiredFreeSpace}, available space: {destinationDriveInfo.TotalFreeSpace}");
             }
 
-            HttpContext.Response.Headers[Constants.Headers.RefreshTopology] = "true";
```
Comment (ppekrol) [inline src/Raven.Server/Web/System/AdminDatabasesHandler.cs:708]: Do we know why it was here?

File: src/Raven.Server/Web/ServerRequestHandler.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.Server/Web/ServerRequestHandler.cs:13]: I think it would be better to store this in memory in OnTopologyChangeInternal. This would avoid us opening transaction for each request and we are already doing that for DatabaseRequestHandler.

Comment (garayx) [reply src/Raven.Server/Web/ServerRequestHandler.cs:13]: done

File: src/Raven.Server/ServerWide/ServerStore.cs [hunk 3]
```diff
@@ -945,6 +950,9 @@ public void NotifyAboutClusterTopologyAndConnectivityChanges()
 
         private void OnTopologyChangeInternal(ClusterTopology topology, Dictionary<string, NodeStatus> status = null)
         {
+            if (_lastClusterTopologyIndex < topology.Etag)
```
Comment (ppekrol) [inline src/Raven.Server/ServerWide/ServerStore.cs:953]: @garayx do we need to read it during ServerStore initialization?

Comment (garayx) [reply src/Raven.Server/ServerWide/ServerStore.cs:953]: done

**Review by ppekrol — APPROVED** [review]
